### PR TITLE
Support cartesian_bounds aggregation on point and shape

### LINKED ISSUE #90157: Bounding Box aggregation for Cartesian points and shapes
### Description

There is absolutely no support for cartesian data in aggregations. There is a meta-task for adding [Cartesian support for aggregations](https://github.com/elastic/elasticsearch/issues/87804), and this issue is the second aggregation to add, the `bounding_box` aggregation.

It is modelled on the existing [`geo_bounding_box`](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-geobounds-aggregation.html) and as such could be called either:

* `cartesian_bounding_box`
* or just `bounding_box`

It is expected to be based on the previous work on https://github.com/elastic/elasticsearch/issues/90156

### PR DESCRIPTION
**Pull request by craigtaverner**
Fixes https://github.com/elastic/elasticsearch/issues/90157

### CODE REVIEW

**Review by iverase — CHANGES_REQUESTED** [review]
I left some minor comments specially for the visibility of `Shapes.ShapeValue`.
Two things:

1) You have some weird file named `jffi4926421121997545579.dylib` which seems some left over.

~~2) We need to check if it is ok to change the API of GeoBounds, not sure if that is an. API we cannot change in a minor.~~
I check things again and everything is clear.

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsIT.java [hunk 1]
```diff
@@ -43,15 +44,15 @@ public void testSingleValuedField() throws Exception {
 
         assertSearchResponse(response);
 
-        GeoBounds geoBounds = response.getAggregations().get(aggName);
+        SpatialBounds<GeoPoint> geoBounds = response.getAggregations().get(aggName);
```
Comment (iverase) [inline server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsIT.java:47]: Can be use here GeoBounds instead?

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsIT.java [hunk 2]
```diff
@@ -69,25 +70,28 @@ public void testSingleValuedField_getProperty() throws Exception {
         assertThat(global.getAggregations(), notNullValue());
         assertThat(global.getAggregations().asMap().size(), equalTo(1));
 
-        GeoBounds geobounds = global.getAggregations().get(aggName);
+        SpatialBounds<GeoPoint> geobounds = global.getAggregations().get(aggName);
```
Comment (iverase) [inline server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsIT.java:73]: The same here

File: x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/metrics/CartesianBoundsAggregationBuilder.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Override
+    protected ValuesSourceRegistry.RegistryKey<?> getRegistryKey() {
+        return REGISTRY_KEY;
+    }
+
+    @Override
+    public Version getMinimalSupportedVersion() {
+        // TODO: Should we have a version here? We only release this for 8.5 or 8.6
+        return Version.V_EMPTY;
```
[the remark this answers is not in the document]
Comment (craigtaverner) [reply x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/metrics/CartesianBoundsAggregationBuilder.java:109]: I noticed that none of the other aggregations had values set here, even when recently added.

[the remark this answers is not in the document]
Comment (craigtaverner) [reply x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/metrics/CartesianBoundsAggregationBuilder.java:109]: OK. I spoke too soon, while the first 4 I looked at did not specify anything, a more careful analysis shows that 4 of the 13 known Geo aggregations did have settings:

* `GeoHexGridAggregationBuilder` -> `Version.V_8_1_0`
* `GeoLineAggregationBuilder` -> `Version.V_7_11_0`
* `GeoGridQueryBuilder` -> `Version.V_8_3_0`
* `GeoTileGridAggregationBuilder` -> `Version.V_7_0_0`

So I will add `Version.V_8_6_0` to the current one.

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsIT.java (hunk 1 above)
Comment (craigtaverner) [reply server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsIT.java:47]: I was in the process of refactoring. I'll finish the job and then you can decide.

File: x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/metrics/CartesianShapeBoundsAggregator.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        if (isNoOp()) {
+            return LeafBucketCollector.NO_OP_COLLECTOR;
+        }
+        CartesianShapeValues values = valuesSource.shapeValues(aggCtx.getLeafReaderContext());
+        return new LeafBucketCollectorBase(sub, values) {
+            @Override
+            public void collect(int doc, long bucket) throws IOException {
+                if (values.advanceExact(doc)) {
+                    maybeResize(bucket);
+                    ShapeValues.ShapeValue value = values.value();
```
[the remark this answers is not in the document]
Comment (craigtaverner) [reply x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/metrics/CartesianShapeBoundsAggregator.java:51]: Done

File: server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsIT.java (hunk 2 above)
Comment (craigtaverner) [reply server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsIT.java]: I've done this for the two dateline tests, but all the rest moved into shared code.

File: x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/SpatialPlugin.java [hunk 5]
```diff
@@ -172,7 +176,15 @@ public List<AggregationSpec> getAggregations() {
...
                 )
-            ).addResultReader(InternalCartesianCentroid::new).setAggregatorRegistrar(this::registerCartesianCentroidAggregator)
+            ).addResultReader(InternalCartesianCentroid::new).setAggregatorRegistrar(this::registerCartesianCentroidAggregator),
+            new AggregationSpec(
+                CartesianBoundsAggregationBuilder.NAME,
+                CartesianBoundsAggregationBuilder::new,
+                usage.track(
+                    SpatialStatsAction.Item.CARTESIANBOUNDS,
+                    CartesianBoundsAggregationBuilder.PARSER    // TODO: Should we have a license for this?
+                )
```
Comment (iverase) [inline x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/SpatialPlugin.java:186]: No, license for CartesianBounds should be the same as for GeoBounds, so it is basic.

**Review by iverase — APPROVED** [review]
LGTM

**Pull request merged**